### PR TITLE
Brgoddar/hackathon/day2

### PR DESCRIPTION
**Pull request by bgoddar**
Add websocket connected/disconnected status messages in the message viewer

### CODE REVIEW

File: packages/devtools-network-console/src/reducers/websocket/index.ts [hunk 1]
```diff
@@ -52,7 +54,12 @@ export default function reduceWebsocket(collection: WS_State = DEFAULT_WS_STATE,
         }
         state = {
             ...state,
-            connected: true
+            connected: true,
+            messages: state.messages.add({
+                direction: 'status',
+                time: 0,
+                content: 'Connected'
+            })
```
Comment (robpaveza) [inline packages/devtools-network-console/src/reducers/websocket/index.ts:62]: Nit: trailing comma (previous line as well)

File: packages/devtools-network-console/src/reducers/websocket/index.ts [hunk 2]
```diff
@@ -79,9 +86,15 @@ export default function reduceWebsocket(collection: WS_State = DEFAULT_WS_STATE,
...
         }
+        // TODO: add disconnected reason/error to content
         state = {
             ...state,
-            connected: false
+            connected: false,
+            messages: state.messages.add({
+                direction: 'status',
+                time: 0,
+                content: 'Disconnected'
```
Comment (robpaveza) [inline packages/devtools-network-console/src/reducers/websocket/index.ts:96]: And here

File: packages/devtools-network-console/src/reducers/websocket/index.ts [hunk 3]
```diff
@@ -79,9 +86,15 @@ export default function reduceWebsocket(collection: WS_State = DEFAULT_WS_STATE,
...
             state = DEFAULT_WS_CONNECTION;
         }
+        // TODO: add disconnected reason/error to content
         state = {
             ...state,
-            connected: false
+            connected: false,
+            messages: state.messages.add({
+                direction: 'status',
+                time: 0,
```
Comment (robpaveza) [inline packages/devtools-network-console/src/reducers/websocket/index.ts:95]: Guess I'll need to add disconnect-time from the host ;)

File: packages/devtools-network-console/src/ui/ResponseViewer/WebSocketMessage.tsx [hunk 4]
```diff
@@ -30,10 +42,10 @@ export default function WebSocketMessage(props: WebSocketMessageProps) {
 
     return (
         <div {...style}>
-            <div {...DESCRIPTOR_STYLE}>
+            {!isStatus ? <div {...DESCRIPTOR_STYLE}>
```
Comment (robpaveza) [inline packages/devtools-network-console/src/ui/ResponseViewer/WebSocketMessage.tsx:45]: For this, so you can omit the fragment, you can use &&, such as this:

```jsx

    {isStatus && (<div {...DESCRIPTOR_STYLE}> 
        ...
    </div>)}
```

File: packages/devtools-network-console/src/reducers/websocket/index.ts (hunk 3 above)
Comment (bgoddar) [reply packages/devtools-network-console/src/reducers/websocket/index.ts:95]: possibly. We aren't currently displaying it in status. I'll make it optional in the Message type, and we can decide if we want to add it later when we handle server disconnect reason

File: packages/devtools-network-console/src/reducers/websocket/index.ts (hunk 1 above)
Comment (bgoddar) [reply packages/devtools-network-console/src/reducers/websocket/index.ts:62]: done

File: packages/devtools-network-console/src/reducers/websocket/index.ts (hunk 3 above)
Comment (bgoddar) [reply packages/devtools-network-console/src/reducers/websocket/index.ts]: put time as an optional param on websocket messages for now.

**Pull request merged**